### PR TITLE
improve DSG performance on findContainedIdentifiers

### PR DESCRIPTION
**Pull request by barshimi**
Close: #6845 

## PR Details
Improve DSG performance stage one: findContainedIdentifiers

### Summary
🕑🚀🐛

This pull request adds performance logging and improves the import logic of the data service generator. It uses console.time, console.timeEnd, and variables to measure and log the execution time of various operations in `build-runner.controller.ts`, `generate-code.ts`, and `create-server.ts`. It also modifies the `create-dto-module.ts` and `ast.ts` files to avoid redundant calls and bugs in the import logic. It also replaces a for loop with a Promise.all call in `create-dtos.ts` to parallelize the creation of the DTO modules.

> _Oh we're the coders of the sea, and we work with speed and skill_
> _We measure and we log the time, of every function and every file_
> _We use `Promise.all` and `AST`, to make our code more fast and neat_
> _And we sing this shanty as we go, to keep our spirits high and sweet_

### Walkthrough
*  Add variables and log statements to measure and log the execution time of various operations in the build runner controller and the data service generator ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17R37), [link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17R43-R45), [link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17R115), [link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17R121-R122), [link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-09c8be16b64619448e37b907c679042b03a2f860056c3f93680f3baf55708cceL83-R94))
*  Add console.time and console.timeEnd calls to measure and log the execution time of the `createDTOs` and `createDTOModules` functions in the `createServerInternal` function of the `create-server.ts` file ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaL58-R64))
*  Replace the for loop with a Promise.all call to parallelize the creation of the DTO modules in the `createDTOModulesInternal` function of the `create-dtos.ts` file and measure and log the execution time of this operation ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-e329d8f984090951f8641118dd4038db0a84644c9c86296cc798ba84497802b0L57-R68))
*  Import and use the `importContainedIdentifiers2` function from the `ast.ts` file in the `create-dto-module.ts` file, which is a modified version of the `importContainedIdentifiers` function that fixes some bugs and improves the performance of importing the identifiers from the DTOs ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-499faa3b11559d2ead6c65f75601db950fe266da18dffa11eb1ed302200c8159R11), [link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-499faa3b11559d2ead6c65f75601db950fe266da18dffa11eb1ed302200c8159L133-R143))
*  Add a variable `importableNamesVar` to store the result of the `getImportableNames` function in the `createDTOModule` function of the `create-dto-module.ts` file and use it later in the `moduleToIds` object to avoid calling the function multiple times ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-499faa3b11559d2ead6c65f75601db950fe266da18dffa11eb1ed302200c8159R107-R108))
*  Add a new function `findContainedIdentifiers2` to the `ast.ts` file, which is a modified version of the `findContainedIdentifiers` function that fixes some bugs and improves the performance of finding the identifiers that are contained in a given AST node ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-95af8698da4339ab1eeb3d8707da727ed0bc613df9b446a3abeca4293c315f42R350-R383))
*  Add a new function `importContainedIdentifiers2` to the `ast.ts` file, which is a modified version of the `importContainedIdentifiers` function that fixes some bugs and improves the performance of importing the identifiers that are contained in a given AST node and uses the `findContainedIdentifiers2` function instead of the `findContainedIdentifiers` function ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-95af8698da4339ab1eeb3d8707da727ed0bc613df9b446a3abeca4293c315f42R468-R507))
*  Remove some whitespace and add some comments in the `importContainedIdentifiers` function of the `ast.ts` file to make the code more readable and consistent ([link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-95af8698da4339ab1eeb3d8707da727ed0bc613df9b446a3abeca4293c315f42L446-R522), [link](https://github.com/amplication/amplication/pull/7358/files?diff=unified&w=0#diff-95af8698da4339ab1eeb3d8707da727ed0bc613df9b446a3abeca4293c315f42L452-R532))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/data-service-generator/src/server/resource/create-dtos.ts [hunk 1]
```diff
@@ -54,17 +54,18 @@ export async function createDTOModulesInternal({
...
-    }
+  console.time(`createDTOModulesInternal-ALL ${dtos.length}`);
+  await Promise.all(
+    entityDTOs.map((dto) => {
+      const isEnumDTO = namedTypes.TSEnumDeclaration.check(dto);
+      return isEnumDTO
+        ? createEnumDTOModule(dto, dtoNameToPath)
+        : createDTOModule(dto, dtoNameToPath);
+    })
+  ).then((modulesRes) => modulesRes.forEach((module) => moduleMap.set(module)));
```
Comment (mulygottlieb) [inline packages/data-service-generator/src/server/resource/create-dtos.ts:65]: Do we need an `await` on the `moduleMap.set()` call? (I am asking since the original code awaited this)

File: packages/data-service-generator/src/server/resource/dto/create-dto-module.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {
   addImports,
   importContainedIdentifiers,
```
Comment (mulygottlieb) [inline packages/data-service-generator/src/server/resource/dto/create-dto-module.ts:9]: please remove the import for the old `importContainedIdentifiers` as it's currently not in use

File: packages/data-service-generator/src/utils/ast.ts [hunk 3]
```diff
@@ -443,15 +517,19 @@ export function importContainedIdentifiers(
...
     const module = idToModule.get(knownId);
     return module;
   });
-  return Object.entries(moduleToContainedIds).map(([module, containedIds]) =>
-    importNames(containedIds, module)
+  const res = Object.entries(moduleToContainedIds).map(
+    ([module, containedIds]) => importNames(containedIds, module)
   );
+
+  return res;
```
Comment (mulygottlieb) [inline packages/data-service-generator/src/utils/ast.ts:532]: Is this change (getting `res` and then returning it instead of just returning the result as in the original code) needed/has any meaning? If not, let's revert this change for clarity.

File: packages/data-service-generator/src/generate-code.ts
Comment (mulygottlieb) [inline packages/data-service-generator/src/generate-code.ts:1]: Please clean all the `console.log()` and `Date.now()` that were used for debugging

File: packages/data-service-generator/src/server/resource/create-dtos.ts
Comment (mulygottlieb) [inline packages/data-service-generator/src/server/resource/create-dtos.ts:1]: Please clean all the `console.log()` and `console.time()` that were used for debugging

**Review by mulygottlieb — APPROVED** [review]

**Comment by overbit** [comment]
@barshimi please link the issue to the PR and fix the CI to be able to merge

**Review by overbit — CHANGES_REQUESTED** [review]
@barshimi please link the issue to the PR and fix the CI to be able to merge

**Review by overbit — APPROVED** [review]
